clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -770,7 +772,8 @@ public void handle(OffsetFetchResponse response, RequestFuture<Map<TopicPartitio
                     log.debug("Group {} failed to fetch offset for partition {}: {}", groupId, tp, error.message());
 
                     if (error == Errors.UNKNOWN_TOPIC_OR_PARTITION) {
-                        future.raise(new KafkaException("Partition " + tp + " may not exist or user may not have Describe access to topic"));
+                        future.raise(new KafkaException("Partition " + tp + " may not exist or the user may not have " +
```

Comment:
I think previously we treated this as retriable because we know the metadata could simply be outdated and so you need to retry in case the topic finally appears. But looking at it from the perspective of auth, we just handle retriable errors by retrying forever. For auth stuff, that may never get resolved... And even in the case of topics that may be getting created, it could just be a fat-fingered config and the topic might never exist. And I think all you'll end up with in this case is debug-level logs that mention the error (if you're lucky enough to have them enabled).

This isn't really a problem with this patch, but something I realized when reviewing it. It might just be the same as https://issues.apache.org/jira/browse/KAFKA-3177 in which case maybe we can defer the solution to that bug, but it seems a likely source of seemingly mysterious issues for users that it might be easy to at least resolve an extra log message into a patch like this to help alleviate the issue.